src/reconstruction/community.jl

```diff
@@ -0,0 +1,388 @@
+    exchange_met_ids::Vector{String};
+    species_name="",
+    biomass_id=""
+    ) where {M<:MetabolicModel}
+
+Add `model` to `community`, which is a pre-existing community model with
+`exchange_rxn_ids` and `exchange_met_ids`. The `species_name` is appended to
+each reaction and metabolite, see [`join`](@ref). If `biomass_id` is specified
+then a biomass metabolite for `model` is also added to the resulting model. The
+column corresponding to the `biomass_id` reaction then produces this new biomass
+metabolite with unit coefficient. Note, `exchange_rxn_ids` and
+`exchange_met_ids` must already exist in the `community` model.
+
+# Example
+```
+community = add_model(community, model, exchange_rxn_ids, exchange_met_ids; species_name="species_2", biomass_id="BIOMASS_Ecoli_core_w_GAM")
+```
+"""
+function add_model(
+    community::CoreModel,
+    model::M,
+    exchange_rxn_ids::Vector{String},
+    exchange_met_ids::Vector{String};
+    species_name = "",
+    biomass_id = "",
+) where {M<:MetabolicModel}
+
+    exchange_met_community_inds = indexin(exchange_met_ids, metabolites(community))
+    exchange_rxn_community_inds = indexin(exchange_rxn_ids, reactions(community))
+    if any(isnothing.(exchange_met_community_inds)) ||
+       any(isnothing.(exchange_rxn_community_inds))
+        throw(
+            DomainError(
+                "exchange metabolite/reaction not found.",
+                "Exchange metabolites/reactions must already be contained in the community model.",
+            ),
+        )
+    end
+
+    n_cmodel_rows, n_cmodel_cols = size(stoichiometry(community))
+    n_model_rows, n_model_cols = size(stoichiometry(model))
+    Iadd, Jadd, Vadd = findnz(stoichiometry(model))
+
+    # shift to fit into community         
+    Iadd .+= n_cmodel_rows
+    Jadd .+= n_cmodel_cols
+
+    # when adding a single model not that many reactions, push! okay?
+    exchange_rxn_model_inds = indexin(exchange_rxn_ids, reactions(model))
+    for i = 1:length(exchange_rxn_ids)
+        isnothing(exchange_rxn_model_inds[i]) && continue
+        push!(Iadd, exchange_met_community_inds[i]) # already present ex met in community model
+        push!(Jadd, n_cmodel_cols + exchange_rxn_model_inds[i]) # new column hence the offset
+        push!(Vadd, 1.0)
+    end
+
+    biomass_met = 0.0
+    if biomass_id != "" # add biomass metabolite
+        biomass_rxn = first(indexin([biomass_id], reactions(model)))
+        push!(Iadd, n_model_rows + n_cmodel_rows + 1)
+        push!(Jadd, biomass_rxn + n_cmodel_cols)
+        push!(Vadd, 1.0)
+        biomass_met = 1
+    end
+
+    n_metabolites_total = n_model_rows + n_cmodel_rows + biomass_met
+    n_reactions_total = n_cmodel_cols + n_model_cols
+
+    I, J, V = findnz(stoichiometry(community))
+    I = [I; Iadd]
+    J = [J; Jadd]
+    V = [V; Vadd]
+    S = sparse(I, J, V, n_metabolites_total, n_reactions_total)
+
+    lbsadd, ubsadd = bounds(model)
+    lbs, ubs = bounds(community)
```

Comment:
wouldn't it be `lb` and `ub` only?

Reply:
So these are arrays of bounds e.g. a bound for each reaction i.e. `lbs = [lb_1, lb_2, ...]` hence the plural

Reply:
can you add a comment?

Reply:
done